### DISCUSSION TITLE
zmodule add homebrew completion with if condition

### DISCUSSION THREAD

**Discussion by cattyhouse** [zimfw/zimfw#477] [Q&A] [answered]
According to [homebrew](https://docs.brew.sh/Shell-Completion)
they have completion files in `$(brew --prefix)/share/zsh/site-functions` 
and the dir has something like this on my system : `_bat    _brew   _fd     _mpv    _rg     _yt-dlp`

so adding in .zimrc
```sh
...
zmodule /opt/homebrew/share/zsh --fpath site-functions
zmodule completion
...
```
 
works like a charm.

But **NOT** this:
```sh
zmodule $(brew --prefix)/share/zsh --fpath site-functions
# zimfw says : no matches found: $(brew --prefix)/share/zsh
```
And **NOT** even this with a if condition:

```sh
if type brew &>/dev/null && [[ $(uname) == Darwin ]] ; then zmodule /opt/homebrew/share/zsh --fpath site-functions ; fi
# not working
```

any suggestions? thanks

EDIT: i wanted to add a if condition because my .zimrc is shared with a bunch of linux machines.

**Comment by antoineco** [comment]
You need to set Homebrew's environment variables _before_ trying what you shared above:

```sh
eval "$(/opt/homebrew/bin/brew shellenv)"
```

I personally have this in my `.zprofile` to ensure it's evaluated before my `.zshrc` (and therefore before Zim).

(source: [Homebrew installation script](https://github.com/Homebrew/install/blob/ba7c27bd650a775430fa9fd42fde54698d4d0570/install.sh#L997-L1037), which explicitly prompts the user to add this line to `.zprofile`)

**Comment by cattyhouse** [comment]
@antoineco thanks, that will export HOMEBREW_PREFIX, so i can use it within .zimrc, that's wonderful. what about the if condition.

**Reply by antoineco** [reply]
It will also update your `PATH` so `type brew` will also work.

**Reply by cattyhouse** [reply]
oh right, i forgot the $path issue. thanks again!

**Comment by cattyhouse** [comment, marked as answer]
finally i come up with a clean and arbitrary if condition without adding the eval to `.zprofile`:

```sh
# in ~/.zimrc
if [[ -d /opt/homebrew/share/zsh ]] && [[ $(uname) == Darwin ]] ; then
    zmodule /opt/homebrew/share/zsh --fpath site-functions
fi
# but before
zmodule completion
```

**Reply by c4shdev** [reply]
- [ ] **Warp** : does not add the completion provided by brew.
- [x] **Terminal & iTerm2** : everything works here. :tada:

## For Intel 

```bash
# in ~/.zimrc
if [[ -d /usr/local/share/zsh/site-functions ]] && [[ $(uname) == Darwin ]] ; then
    zmodule /usr/local/share/zsh/site-functions --fpath .
fi
# but before
zmodule completion
```

## For ARM | M1 ..

```bash
# in ~/.zimrc
if [[ -d /opt/homebrew/share/zsh ]] && [[ $(uname) == Darwin ]] ; then
    zmodule /opt/homebrew/share/site-functions --fpath .
fi
# but before
zmodule completion
```

**Comment by cattyhouse** [comment]
thanks, we can also test `[[ $(uname -m) == arm64 ]]` , true if it's Apple Silicon, false if it's Intel.
